File: 1784. Check if Binary String Has at Most One Segment of Ones/Solution.java
class Solution {
    public boolean checkOnesSegment(String s) {
        int n = s.length();

        if(n==0 || n==1 || n==2){
            return true;
        }
        int i = 0;

        while(i<n && s.charAt(i)=='0'){
            i++;
        }

        while(i<n && s.charAt(i)=='1'){
            i++;
        }

        while(i<n && s.charAt(i)=='0'){
            i++;
        }

        if(i<n && s.charAt(i)=='1'){
            return false;
        }

        return true;
    }
}

File: 1784. Check if Binary String Has at Most One Segment of Ones/Solution.py
class Solution(object):
    def checkOnesSegment(self, s):
        """
        :type s: str
        :rtype: bool
        """
        n = len(s)

        if n==1 or n==2 or n==0:
            return True

        i = 0 

        while i<n and s[i]=='0':
            i+=1

        while i<n and s[i]=='1':
            i+=1

        while i<n and s[i]=='0':
            i+=1
        
        if i<n and s[i]=='1':
            return False
        
        return True
        
        